src/FileType_HDF5.f90

```diff
@@ -1153,6 +1162,109 @@ FUNCTION pathexists_HDF5FileType(thisHDF5File,path) RESULT(bool)
     ENDFUNCTION pathexists_HDF5FileType
 !
 !-------------------------------------------------------------------------------
+!> @brief   Returns whether an HDF5 File or data set has compression
+!> @param   thisHDF5File the HDF5FileType object to interrogate
+!> @param   path (optional) the path to a dataset in the file
+!> @returns bool the logical result of whether the specified object has
+!>          compression.
+!>
+    FUNCTION isCompressed_HDF5FileType(thisHDF5File,path) RESULT(bool)
+      CLASS(HDF5FileType),INTENT(INOUT) :: thisHDF5File
+      CHARACTER(LEN=*),INTENT(IN),OPTIONAL :: path
+      LOGICAL(SBK) :: bool
+#ifdef FUTILITY_HAVE_HDF5
+      INTEGER(SIZE_T),PARAMETER :: namelen=180
+      CHARACTER(LEN=namelen) :: filter_name
+      INTEGER(SIK) :: i,nfilters,filter_id,error,flags,cd_values(1)
+      INTEGER(HID_T) :: dset_id,dcpl
+      INTEGER(SIZE_T) :: nelmts
+      TYPE(StringType) :: path2
+
+      bool=.FALSE.
+
+      ! Make sure the object is initialized, and opened
+      REQUIRE(thisHDF5File%isinit)
+      REQUIRE(thisHDF5File%isOpen())
+
+      IF(PRESENT(path)) THEN
+        IF(pathexists_HDF5FileType(thisHDF5File,path)) THEN
+          path2=convertPath(path)
+          nelmts=1
+
+          !Get the data set ID
+          CALL h5dopen_f(thisHDF5File%file_id,TRIM(path2),dset_id,error)
+
+          !Get the data set creation property list
+          CALL h5dget_create_plist_f(dset_id,dcpl,error)
+
+          !Get the number of filters on the data set, and loop over
+          !them to find a compression filter
+          CALL h5pget_nfilters_f(dcpl,nfilters,error)
+          DO i=0,nfilters-1
+            CALL h5pget_filter_f(dcpl,i,flags,nelmts,cd_values, &
+              namelen,filter_name,filter_id,error)
+            bool=ANY(filter_id == (/H5Z_FILTER_DEFLATE_F, &
+              H5Z_FILTER_SZIP_F,H5Z_FILTER_NBIT_F,H5Z_FILTER_NBIT_F/))
+            IF(bool) EXIT
+          ENDDO
+        ENDIF
+      ELSE
+        bool=thisHDF5File%hasCompression
```

Comment:
Did the HDF5 file type always have the compression attribute?

Reply:
I don't remember, it's had it since we implemented the restart file.